http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -354,6 +406,122 @@ class RuntimeServiceInterp[F[_]: Parallel](blocker: Blocker,
         }
     }
   }
+
+  /**
+   * This method validates an UpdateRuntimeConfigRequest against the current state of the runtime,
+   * and potentially sends a PubSub message (or throws an error). The PubSub receiver does not
+   * do validation; it is assumed all validation happens here.
+   */
+  private[service] def processUpdateRuntimeConfigRequest(req: UpdateRuntimeConfigRequest,
```

Comment:
I think this logic is the same as in `LeonardoService`. I added a lot of unit tests to make sure.

Reply:
The method is quite long. Does it make sense to factor out the `case`s so it's easier to follow at a glance? It also may (or may not) help factoring out some lines that are duplicated.

Reply:
Yeah, I'll see if anything makes sense to factor out. However having a lot of separate methods is not (to me) always easier to follow if the logic is complex.

Reply:
yea. I don't like looking at code with too many unnecessary small functions where I have to jump around a lot

Reply:
I find it easier to look at "table of contents" before diving into the "doc" but I see yous' perspective as well  ☺️